src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,117 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.InvalidDataException;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String EXAMPLE_OF_LOGIN = "Thompson";
+    private static final int EXAMPLE_OF_AGE = 23;
+    private static final int MIN_ALLOWED_AGE = 18;
+
+    private static final String EXAMPLE_OF_PASSWORD = "123456artem";
+    private static final String EXAMPLE_OF_INVALID_PASS = "adqd";
+    private static final int MAX_ALLOWED_AGE = 125;
+    private static final String EMPTY_LINE = "";
+    private static final int NEGATIVE_AGE = -33;
+    private static RegistrationService registrationService;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setAge(EXAMPLE_OF_AGE);
+        user.setLogin(EXAMPLE_OF_LOGIN);
+        user.setPassword(EXAMPLE_OF_PASSWORD);
+    }
+
+    @Test
+    void register_LoginIsNull_notOk() {
+        user.setLogin(null);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_AgeIsNull_NotOk() {
+        user.setAge(null);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_AgeIsLessThanAllowed_NotOk() {
```

Comment:
```suggestion
    void register_ageIsLessThanAllowed_notOk() {
```
likewise in the following cases